doc/boot-requirements.md

```diff
@@ -63,42 +63,49 @@
 			- **only requires to use the existing EFI partition**
 	- with a LVM-based proposal
 		- if there are no EFI partitions
-			- **requires /boot and a new /boot/efi partition**
+			- **requires only a new /boot/efi partition**
 		- if there is already an EFI partition
-			- **requires /boot and a reused /boot/efi partition**
+			- **only requires to use the existing EFI partition**
 - not using UEFI (legacy PC)
-	- with an MS-DOS partition table
-		- with sufficently large MBR gap
+	- with GPT partition table
+		- in a partitions-based proposal
+			- if there is no GRUB partition
+				- **requires a new GRUB partition**
+			- if there is already a GRUB partition
+				- **does not require any particular volume**
+		- in a LVM-based proposal
+			- if there is no GRUB partition
+				- **requires a new GRUB partition**
+			- if there is already a GRUB partition
+				- **does not require any particular volume**
+	- with a MS-DOS partition table
+		- if the MBR gap is big enough to embed Grub
 			- in a partitions-based proposal
 				- **does not require any particular volume**
 			- in a LVM-based proposal
-				- **requires only a /boot partition**
+				- if the MBR gap has additional space for grubenv
+					- **does not require any particular volume**
+				- if the MBR gap has no additional space
+					- **requires only a /boot partition**
 		- with too small MBR gap
-			- **raises an exception**
-		- with no MBR gap
-			- **raises an exception**
-		- with GPT partition table
-			- if there is no GRUB partition
-				- in a partitions-based proposal
-					- **only requires a new GRUB partition**
-				- in a LVM-based proposal
-					- **requires /boot and a GRUB partitions**
-			- if there is already a GRUB partition
-				- in a partitions-based proposal
+			- in a partitions-based proposal
+				- if proposing root (/) as Btrfs
```

Comment:
Hi Steffen,

Thanks for your review. (I think we are talking msdos table on leagacy bios systems)

We fallback to use btrfs bootloader area if mbr gap was too small, due to many laptop/desktop systems shipped with MS Windows are pre-partitioned without enough mbr gap size and our default installation can't work on it. See bsc#879271.

Please note it only works for plain btrfs installation without any lvm and mdadm so that the 64kB size is sufficient. If you have lvm or mdadm, we cannot embed the btrfs area at all because it requires "raw" sectors to work, so even if btrfs offers enough space it still not helpful.

So it's obvious to me. if "map gap too small" error has happened for btrfs on top of lvm/mdadm, we have no way out unless propose a /boot partition for it (with other fs like ext4). 

Hi ancorgs,
I am very sorry for missing out this aspect and the plan should be changed to.

> with too small MBR gap or with no MBR gap at all
>             in a partitions-based proposal
>                 if proposing root (/) as Btrfs
>                     without lvm or mdadm: does not require any particular volume
>                     with lvm or mdadm: require /boot partition with ext4 that supports blocklist.

Is the explanation and plan OK for you and Steffen ?